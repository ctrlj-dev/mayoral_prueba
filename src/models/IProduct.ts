export default interface IProduct {
    id: number,
    name: string,
    price: number,
    discount?: number,
    colors?: string[]
    featuredImg: string
}