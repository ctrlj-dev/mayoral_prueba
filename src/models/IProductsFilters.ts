export default interface IOptions {
    id: number,
    option: string
}

export enum ESortProductsByPrice {
    NOORDER = "1",
    ASC = "2",
    DESC = "3"
}

