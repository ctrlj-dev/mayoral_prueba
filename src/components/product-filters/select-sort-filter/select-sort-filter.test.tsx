import { render } from '@testing-library/react';
import { ESortProductsByPrice } from 'models/IProductsFilters';
import SelectSortFilter from './select-sort-filter-component';

describe('Select Sort Filter specs', () => {
    test('Test to check if components its rendering properly', () => {
      
        const options =
        [
            { id: 1, option: "Ordenar por:" },
            { id: 2, option: "Precio ascendente" },
            { id: 3, option: "Precio descendente" }
        ]

        //Arrange
        //Act
        const screen = render(
            <div data-testid="component">
                <SelectSortFilter
                    id={'sortFilter'}
                    options={options}
                    onChange={(e) => e.target}
                    defaultOption={ESortProductsByPrice.NOORDER}
                />
            </div>)

        const element = screen.getByTestId('component').firstChild.nodeName;

        //Assert
        expect(element).not.toBeNull();
        expect(screen.asFragment()).toMatchSnapshot();

    })
})

