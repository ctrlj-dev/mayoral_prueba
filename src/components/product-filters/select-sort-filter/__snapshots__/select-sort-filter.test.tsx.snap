// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Sort Filter specs Test to check if components its rendering properly 1`] = `
<DocumentFragment>
  <div
    data-testid="component"
  >
    <div
      class="selectSortFilterContainer"
    >
      <select
        id="sortFilter"
      >
        <option
          selected=""
          value="1"
        >
          Ordenar por:
        </option>
        <option
          value="2"
        >
          Precio ascendente
        </option>
        <option
          value="3"
        >
          Precio descendente
        </option>
      </select>
    </div>
  </div>
</DocumentFragment>
`;
